tools/cli/main.c

```diff
@@ -106,10 +106,29 @@ int main(int argc, char **argv)
 
         if (pCmd)
         {
-            if (pCmd->ReqRoot && geteuid())
+            if (pCmd->ReqRoot)
             {
-                dwError = ERROR_TDNF_PERM;
-                BAIL_ON_CLI_ERROR(dwError);
+                if (geteuid())
+                {
+                    dwError = ERROR_TDNF_PERM;
+                    BAIL_ON_CLI_ERROR(dwError);
+                }
+
+                /*
+                 * For operations other than makecache & clean
+                 * suppressing info and prompting y/n doesn't make sense
+                 * Enable info if assumeyes or assumeno is not provided
+                 * along with quiet.
+                 */
+                if ((strcmp(pszCmd, "makecache") &&
```

Comment:
Why exceptions only for `clean` and `makecache`? There are a lot more commands that don't require input. It may be better to have a list of commands that may expect input, that is only commands that install, remove or update/downgrade packages.

Reply:
Good input. Ack.

Reply:
My idea here is:

If a command needs to be run as root and if quiet is set for options other than makecache & clean, then abort.
So, I think this is enough and simple. Can you give me an example to the problem you see here?

Reply:
There are a lot of commands that don't require input, like `check`, `check-local`, etc. So if you do `tdnf check -q` your change will throw an error.

I haven't counted, but I think there are more commands that do *not* require input than there are that do.

Reply:
No; `check` won't throw error because it doesn't need root permissions. Only exception I found is for `tdnf history` rollback, undo, redo. But I think we should switch to simpler solution like I explained below.